bindings/python/pinocchio/visualize/viser_visualizer.py

```diff
@@ -0,0 +1,300 @@
+
+        if self.collisionRootFrame.visible:
+            self.updatePlacements(pin.GeometryType.COLLISION)
+
+        if self.visualRootFrame.visible:
+            self.updatePlacements(pin.GeometryType.VISUAL)
+
+        if self.framesRootFrame.visible:
+            self.updateFrames()
+
+    def displayCollisions(self, visibility):
+        self.collisionRootFrame.visible = visibility
+        self.updatePlacements(pin.GeometryType.COLLISION)
+
+    def displayVisuals(self, visibility):
+        self.visualRootFrame.visible = visibility
+        self.updatePlacements(pin.GeometryType.VISUAL)
+
+    def displayFrames(self, visibility):
+        self.framesRootFrame.visible = visibility
+        self.updateFrames()
+
+    def drawFrameVelocities(self, *args, **kwargs):
+        raise NotImplementedError("drawFrameVelocities is not yet implemented.")
+
+    def updatePlacements(self, geometry_type):
+        if geometry_type == pin.GeometryType.VISUAL:
+            geom_model = self.visual_model
+            geom_data = self.visual_data
+            prefix = self.viewerRootNodeName + "/visual"
+        else:
+            geom_model = self.collision_model
+            geom_data = self.collision_data
+            prefix = self.viewerRootNodeName + "/collision"
+
+        pin.updateGeometryPlacements(self.model, self.data, geom_model, geom_data)
+        for id, geometry_object in enumerate(geom_model.geometryObjects):
+            # Get mesh pose.
+            M = geom_data.oMg[id]
+
+            # Update viewer configuration.
+            frame_name = prefix + "/" + geometry_object.name
+            frame = self.frames[frame_name]
+            frame.position = M.translation * geometry_object.meshScale
+            frame.wxyz = pin.Quaternion(M.rotation).coeffs()[
+                [3, 0, 1, 2]
+            ]  # Pinocchio uses xyzw
+
+    def updateFrames(self):
+        pin.updateFramePlacements(self.model, self.data)
+        for id, frame in enumerate(self.model.frames):
+            # Get frame pose.
+            M = self.data.oMf[id]
+
+            # Update viewer configuration.
+            viser_frame_name = self.framesRootNodeName + "/" + frame.name
+            viser_frame = self.frames[viser_frame_name]
+            viser_frame.position = M.translation
+            viser_frame.wxyz = pin.Quaternion(M.rotation).coeffs()[
+                [3, 0, 1, 2]
+            ]  # Pinocchio uses xyzw
+
+    def captureImage(self, w=None, h=None, client_id=0, transport_format="jpeg"):
+        """
+        Capture an image from the Viser viewer and return an RGB array.
+
+        Parameters:
+            w: The width of the captured image.
+            h: The height of the captured image.
+            client_id: The ID of the Viser client handle.
+            transport_format: The transport format to use for the captured image.
+                Can be "jpeg" (default) or "png".
+        """
+        cli = self.viewer._get_client_handle(client_id)
```

Comment:
If you're fine with any client (or assume one connected client at a time), maybe you could do `next(iter(viewer.get_clients().values()))`?

Reply:
The main issue here is that `client_id=0` will break if you just open the visualizer and refresh; the ID for the second connection would increment to 1

Reply:
Done -- thanks for the suggestion!